specification/common-types/resource-management/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,400 @@
+        }
+      }
+    },
+    "AccessRuleDirection": {
+      "description": "Direction of Access Rule",
+      "enum": [
+        "Inbound",
+        "Outbound"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "AccessRuleDirection",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Inbound",
+            "description": "Applies to inbound network traffic to the secured resources."
+          },
+          {
+            "value": "Outbound",
+            "description": "Applies to outbound network traffic from the secured resources"
+          }
+        ]
+      }
+    },
+    "AccessRuleProperties": {
+      "description": "Properties of Access Rule",
+      "type": "object",
+      "properties": {
+        "direction": {
+          "$ref": "#/definitions/AccessRuleDirection"
+        },
+        "addressPrefixes": {
+          "description": "Address prefixes in the CIDR format for inbound rules",
+          "type": "array",
+          "items": {
+            "description": "An IP address prefix (CIDR) for inbound rules",
+            "type": "string"
+          }
+        },
+        "subscriptions": {
+          "description": "Subscriptions for inbound rules",
+          "type": "array",
+          "items": {
+            "description": "Subscription identifiers",
+            "type": "object",
+            "properties": {
+              "id": {
+                "description": "The fully qualified Azure resource ID of the subscription e.g. ('/subscriptions/00000000-0000-0000-0000-000000000000') ",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "networkSecurityPerimeters": {
+          "description": "Network security perimeters for inbound rules",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeter"
+          }
+        },
+        "fullyQualifiedDomainNames": {
+          "description": "Fully qualified domain names (FQDN) for outbound rules",
+          "type": "array",
+          "items": {
+            "description": "A fully qualified domain name (FQDN)",
+            "type": "string"
+          }
+        },
+        "emailAddresses": {
+          "description": "Email addresses for outbound rules",
+          "type": "array",
+          "items": {
+            "description": "An email address",
+            "type": "string"
+          }
+        },
+        "phoneNumbers": {
+          "description": "Phone numbers for outbound rules",
+          "type": "array",
+          "items": {
+            "description": "A phone number",
+            "type": "string"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "NetworkSecurityPerimeterConfigurationNameParameter": {
+      "name": "networkSecurityPerimeterConfigurationName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "pattern": "^.*$",
```

Comment:
Seems like we should at least have the standard resource restriction, this actually can't be zero length, or contain any character (or longer than 99) I think.

Reply:
Seems like the documented maximum resource name length is a whopping 512 characters. I could put that in.

Reply:
Looking at prior art, like this: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json#L631

It is not common to provide a pattern at all, but is common to provide a minimum and maximum size.

The current pattern provides no restriction, so I think we should remove it.  I am wondering, though, if there aren't any meaningful restrictions , why provide this parameter at all?  What benefit is using this parameter giving to specs?

Reply:
It could at least lead to consistent naming and documentation....